universe/NamedValueRefManager.h

```diff
@@ -79,8 +79,23 @@ struct FO_COMMON_API NamedRef final : public ValueRef<T>
     }
 
     std::string Dump(unsigned short ntabs = 0) const override {
-        auto ref = GetValueRef();
-        return ref ? ref->Dump() : "NAMED_REF_UNKNOWN";
+        std::string retval = "Named";
+        if constexpr (std::is_same<T, int>::value) {
+            retval += "Integer";
+        } else if constexpr (std::is_same<T, double>::value) {
+            retval += "Real";
+        } else {
+            retval += typeid(T).name();
+        }
+        if (m_is_lookup_only) {
+            retval += "Lookup";
+        }
+        retval += " name = \"" + m_value_ref_name + "\"";
+        if (!m_is_lookup_only) {
+            auto ref = GetValueRef();
+            retval += " value = " + (ref ? ref->Dump() : "NAMED_REF_UNKNOWN");
+        }
```

Comment:
I think the "if" partly beats the purpose here. GetValueRef actually looks in the registry if there is anything registered or not and that is a good information i guess.
I suggest that you add something like
else if (! GetValueRef()) retval += " (NAMED_REF_UNKNOWN)";

Reply:
(OTOH this might be out of scope for the dump function)

Reply:
@agrrr3 Yes, I case of Dump we don't need value here because original FOCS script didn't have it.

Reply:
ok, we already log a warning in GetValueRefImpl for unknown value refs. that should suffice